### ISSUE TITLE
Warnings when generating `CHANGELOG.md`

### ISSUE THREAD

**Issue by gian1200** [absolute-version/commit-and-tag-version#195]
**Describe the bug**

2 warnings:
- When running `commit-and-tag-version` on Windows (and having default git configuration), file is generated with LF instead of CRLF.
- According to `markdownlint`, generated `CHANGELOG.md` has an extra (unnecessary) break line. One line before each commit type header is enough. No need to put 2.

**Current behavior**
```shell
npx commit-and-tag-version --first-release
× skip version bump on first release
√ outputting changes to CHANGELOG.md
√ committing CHANGELOG.md
warning: in the working copy of 'CHANGELOG.md', LF will be replaced by CRLF the next time Git touches it

warning: in the working copy of 'CHANGELOG.md', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'CHANGELOG.md', LF will be replaced by CRLF the next time Git touches it
...
```

![image](https://github.com/user-attachments/assets/7a661fbe-c8f3-4d49-b37b-be5caa7ae98b)
![image](https://github.com/user-attachments/assets/c90c065f-51f3-43fc-ad2b-01de827cc078)

**Expected behavior**
No warnings

**Environment**

- `commit-and-tag-version` version(s): v12.5.0
- Node/npm version: Node 20.17.0/npm 10.8.2
- OS: Windows 11

**Possible Solution**

- Read git configuration (`git config --get core.autocrlf`) and generate the file accordingly
    - if OS == Windows and core.autocrlf is true, then CRLF, else LF

- Remove the extra line from generated file

**Additional context**

**Comment by gian1200** [comment]
Update:
After reading the code, it sems that both issues come from [conventional-changelog](https://github.com/conventional-changelog/conventional-changelog).
Also, at least one is already fixed on [conventional-changelog-conventionalcommits](https://github.com/conventional-changelog/conventional-changelog/releases/tag/conventional-changelog-conventionalcommits-v8.0.0) v8.0.0
 - https://github.com/conventional-changelog/conventional-changelog/issues/1188

Currently `commit-and-tag-version` references an old version
https://github.com/absolute-version/commit-and-tag-version/blob/47ab7ded117b935d427eab5a2737b57fdc8c484f/package.json#L40-L45

**Comment by TimothyJones** [comment]
That's right! The linked issue is the right place to report this. I'll leave this issue open, pending their fix (since it will mean we'll need to bump the version here).

On the fixed half of this issue - I'm definitely keen to get in the newer versions of the conventional-changelog libs - I haven't yet because it'll require a close read of the breaking changes. I'm happy to merge it in and release a breaking bump here, but I want to make sure our changelog is right.

I'm not sure how much work it'll be -  it's possible that the very latest will mean we'll need a conversion to ESM (which is worth doing anyway).

**Comment by TimothyJones** [comment]
If anyone has the capacity to look at migrating to the later versions, I'd very happily accept PRs. Otherwise I'll get to it when I can.

**Comment by gian1200** [comment]
I might be able to do it in the next few days/weeks. I also notice that there are other dependencies that need some love, too (e.g. ESLint).

My only question would be regarding contribution guidelines (I didn't see any CONTRIBUTING.md). For instance:

- to not include package.lock.json (for security reasons), and only put package.json in PRs; being the responsibility of maintainers to complete the final step (run npm install)
-to not mix multiple changes on a single PR
- etc.

**Comment by TimothyJones** [comment]
Huh, I thought we had one, but it seems not.

I agree with both of those. Generally it's also:

* If adding / changing a feature please ensure there are tests for the new behaviour
* If there are user facing changes / new features, please ensure the readme is updated
* Please title the PR with conventional commits, as PRs are squash merged - and of course, these become the changelog, so it's best for `fix` / `feat` commits to be titled assuming they're read by users

**Comment by gian1200** [comment]
I tried to upgrade it, but the ESM migration is required on latest versions of `conventional-changelog` packages.

I also tried to upgrade them to latest version compatible with CommonJS. However, I found [some issues with presets](https://github.com/conventional-changelog/conventional-changelog/issues/424) while running tests. I wasn't able to find a fix, so I took the opportunity to work on the following:

- https://github.com/absolute-version/commit-and-tag-version/pull/198
- https://github.com/absolute-version/commit-and-tag-version/pull/199
- https://github.com/absolute-version/commit-and-tag-version/pull/200

These should:
- ease migration to ESM
- get feedback regarding PRs on the repo

**Comment by gian1200** [comment]
#### UPDATE

Preset issues are due to https://github.com/conventional-changelog/conventional-changelog/issues/1308

So the only way to upgrade `conventional-changelog` is to move to ESM or wait for the fix

**Comment by gian1200** [comment]
Hello again! https://github.com/conventional-changelog/conventional-changelog/issues/1308 is closed, and I managed to do a small bump to `conventional-changelog` without the need to migrate to ESM. PR: https://github.com/absolute-version/commit-and-tag-version/pull/246

Migration to ESM seems doable, but the current tests need more love. They are too coupled and I'm not sure if failed tests are due to error in the migration or a bug in the tests.

**Comment by TimothyJones** [comment]
Apologies for the delay, this should be fixed in 13.0.0. Please feel free to reopen if you have further issues.

**Comment by gian1200** [comment]
Awesome! Nice work! 🔥🔥🔥

I just tested it and it indeed solved the "duplicate" issue.
The Windows issue is still open (https://github.com/conventional-changelog/conventional-changelog/issues/1307)

-------------------

However, I found a new bug. When generating a new release, there is no space between the last commit and the previous version title. 

```markdown
...
* a description asdf ([123456](URL...))
## [1.1.0](URL.../compare/v1.1.0...v1.2.0) (a date)
...
```

#### It should be like this:

```markdown
...
* a description asdf ([123456](URL...))

## [1.1.0](URL.../compare/v1.1.0...v1.2.0) (a date)
...
```

### Questions

- Is this a known issue? 
- `commit-and-tag-version` is not the culprit, but `https://github.com/conventional-changelog/conventional-changelog`, right?

**Comment by TimothyJones** [comment]
Yeah, I think that's an upstream issue. I see it too.

Is it a bug, though? It displays correctly in markdown viewers.

**Comment by gian1200** [comment]
It's a lint issue.

`MD032/blanks-around-lists: Lists should be surrounded by blank lines`.

I tried to check if I could send a PR to them, but It seems that they stopped using .hbs templates. I haven't figured out where the fix should be put, yet.

--------

Update: I created the following issue there https://github.com/conventional-changelog/conventional-changelog/issues/1511
